src/Ast.ml

```diff
@@ -35,6 +42,12 @@ module Expr = struct
         (** [let u = expr1 in expr2] *)
     | Letbox of { idm : Id.M.t; boxed : t; body : t }
         (** [letbox u = expr1 in expr2] *)
+    | Match of { matched : t; bound : Id.R.t; alt_empty : t; alt_cons : t }
+        (** FOR NAT ONLY
+          [match name with 
+              | nil => ...
+              | succ bound => ...
+            end] *)
```

Comment:
```suggestion
    | Match of { matched : t; zbranch : t; pred : Id.R.t; sbranch : t }
        (** FOR NAT ONLY
          [match matched with 
              | zero => <zbranch>
              | succ pred => <sbranch>
            end] *)
```
Notice that this is only a very special case of a pattern-matching construction. We are not allowing multiple branches nor nested patterns nor wildcards (`_`). I'm fine with this implementation because it allows us to move forward incrementally, though.